pynamodb/attributes.py

```diff
@@ -506,6 +447,26 @@ def deserialize(self, value):
         return json.loads(value)
 
 
+class NumberSetAttribute(Attribute[Set[float]]):
+    """
+    A number set attribute
+    """
+    attr_type = NUMBER_SET
+    null = True
+
+    def serialize(self, value):
+        """
+        Encodes a set of numbers as a JSON list. Encodes empty sets as "None".
+        """
+        return [json.dumps(v) for v in value] if value else None
+
+    def deserialize(self, value):
+        """
+        Returns a set from a JSON list of numbers.
+        """
+        return {json.loads(v) for v in value}
```

Comment:
Removes the check if value is None to enforce decoding only DynamoDB supported values for the NS type.